gnd/src/main/java/com/google/android/gnd/ui/editobservation/EditObservationViewModel.java

```diff
@@ -220,15 +226,17 @@ private Completable saveBitmapAndUpdateResponse(Bitmap bitmap, Field field) thro
         getRemoteDestinationPath(
             args.getProjectId(), args.getFormId(), args.getFeatureId(), localFileName);
 
-    // TODO: Handle response after reloading view-model and remove this field
-    isPhotoFieldUpdated = true;
-
-    // update observable response map
-    onResponseChanged(field, TextResponse.fromString(destinationPath));
+    Map<Field, String> map = new HashMap<>();
+    map.put(field, destinationPath);
+    photoUpdated.postValue(map);
```

Comment:
`ImmutableMap.of(field, destinationPath)` should work here.

Reply:
Thanks. I didn't think of that!